openmetadata-service/src/main/java/org/openmetadata/service/resources/datamodels/DataModelResource.java

```diff
@@ -0,0 +1,443 @@
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.io.IOException;
+import java.util.UUID;
+import javax.json.JsonPatch;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import org.openmetadata.schema.api.data.CreateDataModel;
+import org.openmetadata.schema.api.data.RestoreEntity;
+import org.openmetadata.schema.entity.data.DataModel;
+import org.openmetadata.schema.type.EntityHistory;
+import org.openmetadata.schema.type.Include;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.DataModelRepository;
+import org.openmetadata.service.jdbi3.ListFilter;
+import org.openmetadata.service.resources.Collection;
+import org.openmetadata.service.resources.EntityResource;
+import org.openmetadata.service.resources.databases.DatabaseUtil;
+import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.util.EntityUtil;
+import org.openmetadata.service.util.RestUtil;
+import org.openmetadata.service.util.ResultList;
+
+@Path("/v1/datamodels")
+@Api(value = "Data Model data asset collection", tags = "Data Model data asset collection")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Collection(name = "datamodels")
+public class DataModelResource extends EntityResource<DataModel, DataModelRepository> {
+  public static final String COLLECTION_PATH = "/v1/datamodels";
+  protected static final String FIELDS = "owner,tags,followers";
+
+  @Override
+  public DataModel addHref(UriInfo uriInfo, DataModel dataModel) {
+    dataModel.setHref(RestUtil.getHref(uriInfo, COLLECTION_PATH, dataModel.getId()));
+    Entity.withHref(uriInfo, dataModel.getOwner());
+    Entity.withHref(uriInfo, dataModel.getService());
+    Entity.withHref(uriInfo, dataModel.getFollowers());
+    return dataModel;
+  }
+
+  public DataModelResource(CollectionDAO dao, Authorizer authorizer) {
+    super(DataModel.class, new DataModelRepository(dao), authorizer);
+  }
+
+  public static class DataModelList extends ResultList<DataModel> {
+    @SuppressWarnings("unused")
+    DataModelList() {
+      // Empty constructor needed for deserialization
+    }
+  }
+
+  @GET
+  @Operation(
+      operationId = "listDataModels",
+      summary = "List dataModels",
+      tags = "dataModels",
```

Comment:
Instead of a separate API section for dataModels, setting tags to `dashboards` will group this under dashboards section in the API.